library/core/src/alloc/layout.rs

```diff
@@ -164,7 +164,8 @@ impl Layout {
     #[unstable(feature = "layout_for_ptr", issue = "69835")]
     pub unsafe fn for_value_raw<T: ?Sized>(t: *const T) -> Self {
         // SAFETY: we pass along the prerequisites of these functions to the caller
-        let (size, align) = unsafe { (mem::size_of_val_raw(t), mem::align_of_val_raw(t)) };
+        let size = size_of_val(t);
+        let align = align_of_val(t);
```

Comment:
There is inconsistency in safety of size_of_val_raw function and intrinsic used to implement it.